t_sne_roots/t_sne_roots.ipynb

```diff
@@ -0,0 +1,156 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "286d2e98-9f33-4e1e-bba4-f526360dfc59",
+   "metadata": {},
+   "source": [
+    "# Interactive t-SNE ROOTS dimensionality reduction"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3d774823",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import datashader as ds\n",
+    "import colorcet as cc\n",
+    "\n",
+    "import holoviews as hv\n",
+    "from holoviews.operation.datashader import datashade, rasterize, dynspread\n",
+    "import panel as pn\n",
+    "\n",
+    "pn.extension()\n",
+    "hv.extension('bokeh')"
```

Comment:
Are both extensions needed? I keep being confused about that, but thought only one was needed.